### PR TITLE
Command Line property expansion and Project target loading

### PR DESCRIPTION
**Pull request by frenchtoast747**
**tl;dr** have a look at the included test to see the scenario.

Assume there are three jamfiles: `Jamroot.jam`, `subdir/build.jam`, and `subdir/subsubdir/build.jam` where each file is a parent to the following. 

Assume that I have a feature declared in `subdir/build.jam`:

```
import feature ;
feature.feature my-feature : : free ;
```

And assume that I have some target to build in the grandchild jamfile, `subdir/subsubdir/build.jam`

```
exe hello : hello.c ;
```

If I try building that target from the root while also specifying the feature declared in `subdir/build.jam`, I will get a feature validation error:

```
$ b2 subdir/subsubdir my-feature="some free value"
error: unknown feature "<my-feature>"
C:/boost-build/src/share/boost-build/src/build\feature.jam:359: in expand-subfeatures-aux from module feature
C:/boost-build/src/share/boost-build/src/build\feature.jam:424: in feature.expand-subfeatures from module feature
C:/boost-build/src/share/boost-build/src/build\build-request.jam:20: in apply-to-property-set from module build-request
(builtin):-1: in sequence.transform from module sequence
C:/boost-build/src/share/boost-build/src/build\build-request.jam:32: in build-request.expand-no-defaults from
module build-request
C:/boost-build/src/share/boost-build/src\build-system.jam:594: in load from module build-system
C:/boost-build\src\share\boost-build\src\kernel\modules.jam:289: in import from module modules
C:/boost-build\src\share\boost-build\src\kernel\bootstrap.jam:139: in boost-build from module
C:/boost-build\src\share\boost-build\boost-build.jam:8: in module scope from module
```

This change delays the property validation and expansion as late as possible to ensure that all projects/modules have been loaded.

### CODE REVIEW

**Comment by vprus** [comment]
Aaron,

thanks for the patch, I think it's overall good. It appears to make unit_tests.py fail - is this something you can fix?

**Comment by frenchtoast747** [comment]
Yes, I will fix the unit tests for build-request.jam and push a fix soon. Thanks!

**Comment by frenchtoast747** [comment]
Since the functionality of `from-command-line` was split, I've updated the unit tests to reflect that.

**Comment by vprus** [comment]
Thanks, cherry-picked to develop.

**Comment by frenchtoast747** [comment]
Thank you!
